spec/components/search_spec.rb

```diff
@@ -1921,4 +1921,37 @@ def indexed_post(*args)
       expect(Search.new("advanced order:chars").execute.posts).to eq([post0, post1])
     end
   end
+
+  context 'search all but topics/posts' do
+    before { SiteSetting.tagging_enabled = true }
+    let!(:user) { Fabricate(:user) }
+    fab!(:group) { Fabricate(:group, name: 'bruce-world-fans') }
+    fab!(:topic) { Fabricate(:topic, title: 'Bruce topic not a result') }
+
+    it 'works' do
+      category = Fabricate(:category_with_definition, name: 'bruceland', user: user)
+      tag = Fabricate(:tag, name: 'brucealicious')
+
+      result = Search.execute('bruce', type_filter: 'exclude_topics')
+
+      expect(result.users.length).to eq(1)
+      expect(result.users[0].id).to eq(user.id)
+
+      expect(result.categories.length).to eq(1)
+      expect(result.categories[0].id).to eq(category.id)
+
+      expect(result.groups.length).to eq(1)
+      expect(result.groups[0].id).to eq(group.id)
+
+      expect(result.tags.length).to eq(1)
+      expect(result.tags[0].id).to eq(tag.id)
```

Comment:
```suggestion
      expect(result.users.map(&:id)).to contain_exactly(user.id)

      expect(result.categories.map(&:id)).to eq(category.id)

      expect(result.groups.map(&:id)).to contain_exactly(group.id)

      expect(result.tags.map(&:id)).to eq(tag.id)
```

Minor but we can assert for both count and accuracy in a single assertion.